components/infobox/wikis/tft/infobox_league_custom.lua

```diff
@@ -0,0 +1,115 @@
+--
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local League = Lua.import('Module:Infobox/League', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+
+local _args
+local lang = mw.language.getContentLanguage()
+
+local CustomLeague = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local GAME_MODES = {
+	solo = 'solos',
+	duo = 'duos',
+	squad = 'squads',
+}
+local UNKNOWN_MODE = 'Unknown'
+local RIOT_ICON = '[[File:Riot Games Tier Icon.png|x12px|link=Riot Games|Tournament supported by Riot Games]]'
+
+function CustomLeague.run(frame)
+	local league = League(frame)
+	_args = league.args
+
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
+	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
+	league.liquipediaTierHighlighted = CustomLeague.liquipediaTierHighlighted
+	league.appendLiquipediatierDisplay = CustomLeague.appendLiquipediatierDisplay
+	league.getWikiCategories= CustomLeague.getWikiCategories
+
+	return league:createInfobox(frame)
+end
+
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local args = _args
+	table.insert(widgets, Cell{
+		name = 'Teams',
+		content = {args.team_number}
+	})
+	table.insert(widgets, Cell{
+		name = 'Players',
+		content = {args.participants_number}
+	})
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		table.insert(widgets, Cell{
+			name = 'Patch',
+			content = {CustomLeague:_createPatchCell(_args)}
+		})
+		table.insert(widgets, Cell{
+			name = 'Game Mode',
+			content = {lang:ucfirst(CustomLeague._getGameMode() or '')}
+		})
+	end
+
+	return widgets
+end
+
+function CustomLeague:liquipediaTierHighlighted(args)
+	return Logic.readBool(args['riot-sponsored'])
+end
+
+function CustomLeague:appendLiquipediatierDisplay()
+	if Logic.readBool(_args['riot-sponsored']) then
+		return ' ' .. RIOT_ICON
+	end
+	return ''
+end
+
+function CustomLeague:defineCustomPageVariables()
+	Variables.varDefine('tournament_mode', CustomLeague._getGameMode())
+end
+
+function CustomLeague:_createPatchCell(args)
+	if String.isEmpty(args.patch) then
+		return nil
+	end
+
+	local content = '[[Patch ' .. args.patch .. '|'.. args.patch .. ']]'
+
+	if String.isEmpty(args.epatch) then
+		return content
+	end
+
+	return content .. ' &ndash; [[Patch ' .. args.epatch .. '|'.. args.epatch .. ']]'
+end
+
+function CustomLeague._getGameMode()
+	return GAME_MODES[string.lower(_args.mode or ''):gsub('s$', '')]
+end
+
+function CustomLeague.getWikiCategories(args)
+	return {(CustomLeague._getGameMode() or UNKNOWN_MODE) .. ' Mode Tournaments'}
```

Comment:
```suggestion
	return {lang:ucfirst(CustomLeague._getGameMode() or UNKNOWN_MODE) .. ' Mode Tournaments'}
```

Reply:
I had this in originally, but even without it the first letter of the category name is always capitalised. Should I add it back in to be on the safe side?

Reply:
Yes please